driftbase/lobbies.py

```diff
@@ -0,0 +1,757 @@
+
+                    new_host_player_id = sorted_members[0]["player_id"]
+
+                    log.info(f"Player {new_host_player_id} promoted to lobby host for lobby '{lobby_id}'")
+
+                    lobby["members"] = sorted_members
+
+                lobby_lock.lobby = lobby
+
+                # Notify remaining members
+                _post_lobby_event_to_members(receiving_player_ids, "LobbyMemberLeft", {"lobby_id": lobby_id, "left_player_id": player_id, "members": lobby["members"]})
+            else:
+                # No one left in the lobby, delete the lobby
+
+                log.info(f"No one left in lobby '{lobby_id}'. Lobby deleted.")
+
+                lobby_lock.lobby = None
+        else:
+            log.error(f"Lobby member player '{player_id}' attempted to leave lobby '{lobby_id}' without being a member")
+
+def _internal_delete_lobby(player_id: int, lobby_id: str):
+    """
+    Caller is responsible for handling the player lobby id lock value
+    """
+    with _LockedLobby(_get_lobby_key(lobby_id)) as lobby_lock:
+        lobby = lobby_lock.lobby
+
+        if not lobby:
+            log.warning(f"Player '{player_id}' attempted to delete lobby '{lobby_id}' which doesn't exist")
+            return
+
+        host_player_id = _get_lobby_host_player_id(lobby)
+
+        if host_player_id != player_id:
+            log.warning(f"Player '{player_id}' attempted to delete lobby '{lobby_id}' without being the host")
+            raise InvalidRequestException(f"You aren't the host of lobby {lobby_id}. Only the lobby host can delete the lobby")
+
+        log.info(f"Lobby host player '{player_id}' deleted lobby '{lobby_id}'")
+
+        for member in lobby["members"]:
+            if not member["host"]:
+                with _GenericLock(_get_player_lobby_key(member["player_id"])) as member_lobby_id_lock:
+                    member_lobby_id_lock.value = None
+
+        # Delete the lobby
+        lobby_lock.lobby = None
+
+        # Notify members
+        receiving_player_ids = _get_lobby_member_player_ids(lobby, [player_id])
+        if receiving_player_ids: # Potentially empty if the host is alone in the lobby
+            _post_lobby_event_to_members(receiving_player_ids, "LobbyDeleted", {"lobby_id": lobby_id})
+
+def _lobby_match_initiated(lobby: dict) -> bool:
+    return lobby["status"] in ("starting", "started")
+
+def _can_join_team(lobby: dict, team: str) -> bool:
+    team_count = 0
+    team_capacity = lobby["team_capacity"]
+    for member in lobby["members"]:
+        team_name = member["team_name"]
+        if team_name == team:
+            team_count += 1
+
+    return team_count < team_capacity
+
+def _get_lobby_member(lobby: dict, player_id: int) -> typing.Optional[dict]:
+    return next((member for member in lobby["members"] if member["player_id"] == player_id), None)
+
+def _generate_lobby_id() -> str:
+    lobby_id_length = _get_tenant_config_value("lobby_id_length")
```

Comment:
Maybe... probably not though.

I placed it as tenant config value since lobby popularity can vary from game to game. One game might want to decrease this for human readable reasons while another might want to increase it for increased range of possibilities due to popular demand.

It does smell a bit of YAGNI, I agree.

Want me to remove it from the tenant config?

Reply:
Removed from tenant config.